### PR TITLE
fix: add translations for hcv section 8 voucher

### PR DESCRIPTION
**Pull request by ludtkemorgan**
This PR addresses #4113 

- [x] Addresses the issue in full
- [ ] Addresses only certain aspects of the issue

## Description

This pulls the translation strings from HBA for the HCV/Section 8 Voucher. This Reserved community type is only used in Alameda at this point, but adding it to core will help for future jurisdictions

Also, there was a failing test unrelated to this change. A test was added to main while a PR was in review and then that PR merged made the test fail. I fixed the null pointer spot but also updated the test to have some preference data so that it better tests the page and these fields.

### CODE REVIEW

File: sites/public/__tests__/pages/applications/preferences/all.test.tsx [hunk 1]
```diff
@@ -2,6 +2,14 @@ import React from "react"
 import { setupServer } from "msw/lib/node"
 import { mockNextRouter, render } from "../../../testUtils"
 import ApplicationPreferencesAll from "../../../../src/pages/applications/preferences/all"
+import { AppSubmissionContext } from "../../../../src/lib/applications/AppSubmissionContext"
```
Comment (mcgarrye) [inline sites/public/__tests__/pages/applications/preferences/all.test.tsx:5]: nit: imports aren't alphabetized

**Review by mcgarrye — APPROVED** [review]
LGTM

File: sites/public/__tests__/pages/applications/preferences/all.test.tsx (hunk 1 above)
Comment (ludtkemorgan) [reply sites/public/__tests__/pages/applications/preferences/all.test.tsx:5]: Is that a new standard we are following? I typically don't alphabetize the imports, but I can start doing that if we want to.

Comment (mcgarrye) [reply sites/public/__tests__/pages/applications/preferences/all.test.tsx:5]: Idk if we ever formalized it but I know Emily and I have been doing that lately. This is fine to go out and we can discuss later

Comment (ludtkemorgan) [reply sites/public/__tests__/pages/applications/preferences/all.test.tsx:5]: Sounds good!

**Pull request merged**